docs/contrib/code-conventions.md

```diff
@@ -0,0 +1,39 @@
+# Code conventions
+
+- [POSIX shell](#posix-shell)
+- [Go](#go)
+- [Directory and file conventions](#directory-and-file-conventions)
+- [Testing conventions](#testing-conventions)
+
+## POSIX shell
+
+- [Style guide](https://google.github.io/styleguide/shell.xml)
+- Ensure that build, release, test, and cluster-management scripts run on macOS
+
+## Go
+
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+- [Effective Go](https://golang.org/doc/effective_go.html)
+- Know and avoid [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
+- Comment your code.
+  - [Go's commenting conventions](http://blog.golang.org/godoc-documenting-go-code)
+  - If reviewers ask questions about why the code is the way it is, that's a sign that comments might be helpful.
+- Command-line flags should use dashes, not underscores
+- Naming
+  - Please consider package name when selecting an interface name, and avoid redundancy. For example, `storage.Interface` is better than `storage.StorageInterface`.
+  - Do not use uppercase characters, underscores, or dashes in package names.
+  - Please consider parent directory name when choosing a package name. For example, `pkg/controllers/autoscaler/foo.go` should say `package autoscaler` not `package autoscalercontroller`.
+    - Unless there's a good reason, the `package foo` line should match the name of the directory in which the `.go` file exists.
+    - Importers can use a different name if they need to disambiguate.
+
+## Directory and file conventions
+
+- Avoid general utility packages. Packages called "util" are suspect. Instead, derive a name that describes your desired function. For example, the utility functions dealing with waiting for operations are in the `wait` package and include functionality like `Poll`. The full name is `wait.Poll`.
+- All filenames should be lowercase.
+- Go source files and directories use underscores, not dashes.
+  - Package directories should generally avoid using separators as much as possible. When package names are multiple words, they usually should be in nested subdirectories.
+- Document directories and filenames should use dashes rather than underscores.
```

Comment:
I don't think this makes sense with our current documentation. For both K3s and RKE2 docs we mainly use underscores in the filenames. The one exception seems to be the `developer-docs` folder itself.

Reply:
Agree and you got me there. I took most of the code conventions from upstream, and I missed that one.

So how should I address it, state the opposite (underscore over dashes) or remove this line?

Reply:
I would just remove the line. And maybe change line 33 to say "All files should use underscores" so we include both go and documentation files.